calico_kubernetes/tests/kube_plugin_test.py

```diff
@@ -416,42 +416,45 @@ def test_delete_docker_interface(self):
             ], any_order=True)
 
     def test_configure_profile(self):
-        with patch_object(self.plugin, '_datastore_client',
-                          autospec=True) as m_datastore_client, \
-                patch_object(self.plugin, '_get_namespace_tag',
-                             autospec=True) as m_get_namespace_tag, \
-                patch_object(self.plugin, '_get_pod_config',
-                             autospec=True) as m_get_pod_config, \
-                patch_object(self.plugin, '_apply_rules',
-                             autospec=True) as m_apply_rules, \
-                patch_object(self.plugin, '_apply_tags',
-                             autospec=True) as m_apply_tags:
-            # Set up mock objects
-            m_datastore_client.profile_exists.return_value = False
-            m_get_pod_config.return_value = 'pod'
-            m_get_namespace_tag.return_value = 'tag'
-
-            # Set up class members
-            pod_name = 'pod_name'
-            profile_name = 'profile_name'
-            self.plugin.pod_name = pod_name
-            self.plugin.profile_name = profile_name
-
-            # Set up args
-            endpoint = Endpoint(TEST_HOST, TEST_ORCH_ID, '1234', '5678',
-                                'active', 'mac')
+        # Set up mock objects
+        m_datastore_client = Mock()
```

Comment:
After thinking more about this suggestion (Alex also didn't spot the need to do `spec=`), I am on the fence as to whether this new way is safer than just using patch.object as a decorator with `autospec=True`.

I think it's about the same number of characters, so I'll leave it up to your preference. I do like the saving of indentation though, so the other option would be to use patch.object as a decorator instead of a context manager.